client/my-sites/earn/customers/cancel-dialog.tsx

```diff
@@ -61,25 +75,30 @@ function CancelDialog( { subscriberToCancel, setSubscriberToCancel }: CancelDial
 	}
 
 	return (
-		<Dialog
-			isVisible={ Boolean( subscriberToCancel ) }
-			buttons={ [
-				{
-					label: translate( 'Back' ),
-					action: 'back',
-				},
-				{
-					label: getText( subscriberToCancel ).button,
-					isPrimary: true,
-					action: 'cancel',
-				},
-			] }
-			onClose={ onCloseCancelSubscription }
-		>
-			<h1>{ translate( 'Confirmation' ) }</h1>
-			<p>{ getText( subscriberToCancel ).confirmation_subheading }</p>
-			<Notice text={ getText( subscriberToCancel ).confirmation_info } showDismiss={ false } />
-		</Dialog>
+		<>
+			{ subscriberToCancel && (
+				<Dialog
+					className="earn__cancel-dialog"
+					isVisible={ Boolean( subscriberToCancel ) }
+					buttons={ [
+						{
+							label: translate( 'Back' ),
+							action: 'back',
+						},
+						{
+							label: getText( subscriberToCancel ).button,
+							isPrimary: true,
+							action: 'cancel',
+						},
+					] }
+					onClose={ onCloseCancelSubscription }
+				>
+					<h1>{ getText( subscriberToCancel ).confirmation_subheading }</h1>
+					<p>{ getText( subscriberToCancel ).confirmation_info }</p>
+					<p>{ getText( subscriberToCancel ).confirmation_info_2 }</p>
+				</Dialog>
+			) }
+		</>
```

Comment:
Code is the same here except wrapped in fragment and conditional. Needed the extra conditional to avoid flash/extra loading of the dialog when closing.